Algorithms/Graphs/Detect_Cycle_In_Undirected_Graph.cpp

```diff
@@ -0,0 +1,71 @@
+// C++ program to detect cycle in an undirected graph using BFS. 
+#include <bits/stdc++.h> 
+using namespace std; 
+
+void addEdge(vector<int> adj[], int u, int v) 
+{ 
+	adj[u].push_back(v); 
+	adj[v].push_back(u); 
+} 
+
+bool isCyclicConntected(vector<int> adj[], int s, 
+						int V, vector<bool>& visited) 
+{ 
+	// Set parent vertex for every vertex as -1. 
+	vector<int> parent(V, -1); 
+
+	// Create a queue for BFS 
+	queue<int> q; 
+
+	// Mark the current node as visited and enqueue it 
+	visited[s] = true; 
+	q.push(s); 
+
+	while (!q.empty()) { 
+
+		// Dequeue a vertex from queue and print it 
+		int u = q.front(); 
+		q.pop(); 
```

Comment:
What is q?

Reply:
q is a queue. It is stated in comment at line 17